Boyer Moore Algorithm in Java

aeyshubh opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I am one of the students selected for Woc and this is my first PR into Neo Algo

### Checklist

-   [x] I've read the contribution guidelines.
-   [x] I've checked the issue list before deciding what to submit.
-   [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1439

aeyshubh committed:
Adding Moore Algorithm .

This is my first Contribution to Neo Algo as part of my Woc.
This code supports Dynamic Input from the user .

aeyshubh committed:
Merge pull request #1 from ShubhamPatel33/ShubhamPatel33-patch-1

Adding Moore Algorithm .

aeyshubh committed:
Added Moore Algorithm to the Readme (#2)

aeyshubh committed:
Rename Moore Algorithm.java to Moore_Algorithm.java (#3)

aeyshubh committed:
Updated the name (#4)

aeyshubh commented:
@iamrajiv sir please accept my PR and let me know if any changes are required

iamrajiv commented:
@ShubhamPatel33 Please provide the issue number for this PR. I there is no issue made on Moore Algorithm then create one so that others can also contribute to this with different languages.

aeyshubh commented:
Okay Rajiv .

On Thu, 10 Dec 2020, 1:00 pm Rajiv Singh, <notifications@github.com> wrote:

> @ShubhamPatel33 <https://github.com/ShubhamPatel33> Please provide the
> issue number for this PR. I there is no issue made on Moore Algorithm then
> create one so that others can also contribute to this with different
> languages.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/TesseractCoding/NeoAlgo/pull/1433#issuecomment-742306312>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMA32UJYFV7G62V6TTZ3LNLSUB2INANCNFSM4UTCAIVQ>
> .
>

aeyshubh commented:
Issue Number : #1439

aeyshubh committed:
Added Moore Algorithm in Python

I have solved the issue #1439 in which i have written code in python for Moore Algorithm.

aeyshubh committed:
Merge pull request #5 from ShubhamPatel33/ShubhamPatel33-patch-1

Added Moore Algorithm in Python

aeyshubh committed:
Added Moore's Algorithm into Readme [issue#1439] (#6)

aeyshubh commented:
Hello,
My code for Moore's Algorithm is ready ,kindly review it and let me know if any changes are required @iamrajiv @plazzy99 .

iamrajiv commented on Python/other/Moore_Algorithm.py:
```diff
@@ -0,0 +1,37 @@
+## This Program Checks the maximum occurence of a number in a array/list
```
@ShubhamPatel33 Format your code according to PEP 8 rules. You can refer to this http://pep8online.com/.

aeyshubh replied:
Changes Made in the file

aeyshubh committed:
Updated RED changes suggested by  PEP8 (#7)

iamrajiv committed:
formatted code

iamrajiv committed:
formatted code

HarshCasper commented on Java/Algorithms/Moore_Algorithm.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.Scanner;
```
Add a Multi-Line Comments explaining the Algorithm in a few lines. Nothing fancy but it will help people to understand the main pitch of the algorithm.

HarshCasper commented on Java/Algorithms/Moore_Algorithm.java:
```diff
@@ -0,0 +1,43 @@
+		for(int i=0;i<arr.length-1;i++) {
+			count=1;
+			for(int j=i+1;j<arr.length;j++) {
+				if(arr[j]==arr[i]) {              // Counting the number by matching it with
```
Keep the Comments above the Line of Code not beside it

HarshCasper commented on Java/Algorithms/Moore_Algorithm.java:
```diff
@@ -0,0 +1,43 @@
+				}
+			}
+			if(count>arr.length/2) {
+				System.out.println("Number Occuring More than 'n/2'="+arr[i]);
```
Format the Code properly.

HarshCasper commented on Java/Algorithms/Moore_Algorithm.java:
```diff
@@ -0,0 +1,43 @@
+	}
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		System.out.println("Enter Number of elements (Eg. 6) : ");
```
Don't use shorthands. We cannot expect everyone to understand them.

aeyshubh replied:
Okay , I will make the changes

aeyshubh replied:
@HarshCasper I have updated the file ,please let me know if any changes are required.

HarshCasper reviewed (changes requested):
Follow the same with your rest implementation as well

aeyshubh committed:
Updated the java code with the changes proposed (#1439)

vatsalkesarwani12 commented on Java/Algorithms/Moore_Algorithm.java:
```diff
@@ -0,0 +1,50 @@
+5
+4
+2
+Number Occuring More than 'n/2'=5
```
Add time and space complexity

aeyshubh replied:
Okay , i will do that

vatsalkesarwani12 commented on Python/other/Moore_Algorithm.py:
```diff
@@ -0,0 +1,29 @@
+def counting_fun(array):
```
Comment about the problem description here also

aeyshubh replied:
Okay

vatsalkesarwani12 commented on Python/other/Moore_Algorithm.py:
```diff
@@ -0,0 +1,29 @@
+# 1
+# 4
+# 2
+# Number occuring more than n/2 is: 1
```
Add time and space complexity

aeyshubh replied:
Okay

aeyshubh committed:
Added description ,Time and space complexity. (#10) in

aeyshubh committed:
Added Time and space complexity (#9)

aeyshubh commented:
I have updated both the files @plazzy99 and @HarshCasper .

vatsalkesarwani12 reviewed (approved):
👍🏻

HarshCasper commented:
LGTM!
